mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/presenter/HistoryPresenter.java

```diff
@@ -90,6 +95,37 @@ public void handleTransactionClick(int transactionIndex) {
         mHistoryView.showTransactionDetailDialog(transactionIndex, accountNumber);
     }
 
+    public List<Transaction> getFilteredTransactions(List<Transaction> transactions,
```

Comment:
IMO the filtering can be made part of code usecase. Instead of adding a new `getFiilteredTransactions`, pass a filterList argument in existing `fetchTransactions` and then make filtering a part of the core usecase.

Let me know if I should provide some more clarity regarding this.